### PR TITLE
Rendering an object for given angles

### PR DESCRIPTION
**Pull request by Lucaweihs**
Adding code to let us more easily render an object from multiple angles at runtime.

### CODE REVIEW

File: unity/Assets/Scripts/BaseFPSAgentController.cs [hunk 1]
```diff
@@ -7405,6 +7405,193 @@ ProceduralAsset asset
...
+            renderCamera.transform.position = originalPosition;
+            renderCamera.transform.rotation = originalRotation;
+            renderCamera.targetTexture = originalRenderTexture;
+            RenderTexture.active = null;
+            RenderTexture.Destroy(renderTexture);
+
+            actionFinished(byte_arrays != null, byte_arrays);
+        }
+
+        public void RenderObjectFromAngles(
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/BaseFPSAgentController.cs:7573]: In the new system you can just make the return type `IEnumerator` and return a `yield new ActionFinished(...)` at some point and the Coroutine will be started for you! so no need to lauch a Coroutine Manually (which can lead to bugs performance issues)

**Comment by AlvaroHG** [comment]
LGTM

File: unity/Assets/Scripts/BaseFPSAgentController.cs (hunk 1 above)
Comment (AlvaroHG) [reply unity/Assets/Scripts/BaseFPSAgentController.cs:7573]: In this case you can also return a call to your other function which is an `IEnumerator` `renderObjectFromAngles`

Comment (Lucaweihs) [reply unity/Assets/Scripts/BaseFPSAgentController.cs:7573]: From what I could tell, this ended up being more complicated than just leaving it as it is (C# doesn't seem to have a nice way to yield from another function call). Kept as it is for now.

**Pull request merged**